library/pool.c

```diff
@@ -276,10 +276,28 @@ static inline void make_thread_stat_id(char *buffer, int size, struct dnet_work_
 	snprintf(buffer, size - 1, "%s.%s", pool->pool_id, mode_marker);
 }
 
+struct dnet_cmd *n2_io_req_access_cmd(struct dnet_io_req *r);
+
+struct dnet_cmd *dnet_io_req_access_cmd(struct dnet_io_req *r) {
+	if (r->io_req_type == DNET_IO_REQ_OLD_PROTOCOL)
+		return r->header; // dnet_io_req enqueued from old mechanic
+	else
+		return n2_io_req_access_cmd(r); // dnet_io_req enqueued from protocol-independent mechanic
+}
+
+struct dnet_cmd *n2_io_req_access_request_cmd_inplace(struct dnet_io_req *r);
```

Comment:
Why you didn't write implementation here?

Reply:
It's implemented on C++, current file is C.

Reply:
Declaration moved to header.